include/reactor-sdk/Reactor.hh

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include "reactor-cpp/reactor-cpp.hh"
+#include <string.h>
+#include "BaseTrigger.hh"
+#include "SystemParameterBase.hh"
+#include "Environment.hh"
+
+namespace sdk
+{
+
+template <typename InputTuple>
+class ReactionInput;
+
+template <typename Fn, typename InputTuple, typename DependencyTuple, typename OutputTuple>
+class Reaction;
+
+template<typename T>
+class Input;
+
+template<typename T>
+class Output;
+
+template<typename T>
+class MultiportOutput;
+
+template<typename T>
+class MultiportInput;
+
+class Timer;
+
+template <typename T>
+struct trigger_value_type;
+
+template <typename T>
+struct trigger_value_type<reactor::LogicalAction<T> *>
+{
+    using type = reactor::LogicalAction<T>&;
+};
+
+template <typename T>
+struct trigger_value_type<Input<T> *>
+{
+    using type = Input<T>&;
+};
+
+template <typename T>
+struct trigger_value_type<MultiportInput<T> *>
+{
+    using type = MultiportInput<T>&;
+};
+
+template <typename T>
+struct trigger_value_type<MultiportOutput<T> *>
+{
+    using type = MultiportOutput<T>&;
+};
+
+template <typename T>
+struct trigger_value_type<Output<T> *>
+{
+    using type = Output<T>&;
+};
+
+template <>
+struct trigger_value_type<reactor::StartupTrigger *>
+{
+    using type = reactor::StartupTrigger&;
+};
+
+template <>
+struct trigger_value_type<reactor::ShutdownTrigger *>
+{
+    using type = reactor::ShutdownTrigger&;
+};
+
+template <>
+struct trigger_value_type<Timer *>
+{
+    using type = Timer&;
+};
+
+class Reactor : public reactor::Reactor
+{
+protected:
+    reactor::StartupTrigger startup{"startup", this};
+    reactor::ShutdownTrigger shutdown{"shutdown", this};
+
+private:
+    size_t bank_index_ = 0;
+    SystemParameterBase *p_param = nullptr;
+    Environment *env{nullptr};
```

Comment:
This is duplicate. You can get it via the `environment()` function.

/env/env_/ (https://google.github.io/styleguide/cppguide.html)